libsolidity/codegen/ir/IRLValue.cpp

```diff
@@ -0,0 +1,103 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * Generator for code that handles LValues.
+ */
+
+#include <libsolidity/codegen/ir/IRLValue.h>
+
+#include <libsolidity/codegen/ir/IRGenerationContext.h>
+#include <libsolidity/codegen/YulUtilFunctions.h>
+#include <libsolidity/ast/AST.h>
+
+#include <libdevcore/Whiskers.h>
+
+using namespace std;
+using namespace dev;
+using namespace dev::solidity;
+
+IRLocalVariable::IRLocalVariable(
+	ostream& _code,
+	IRGenerationContext& _context,
+	VariableDeclaration const& _varDecl
+):
+	IRLValue(_code, _context, _varDecl.annotation().type),
+	m_variableName(_context.localVariableName(_varDecl))
+{
+}
+
+void IRLocalVariable::storeValue(string const& _value, Type const& _type) const
+{
+	solAssert(_type == *m_type, "Storing different types - not necessarily a problem.");
```

Comment:
Is the type conversion supposed to happen before calling this function?

Reply:
Yes, I have the impression that the type checker ensures that.